_posts/2022-03-25-emergent-symbols.md

```diff
@@ -0,0 +1,708 @@
+unlike any of the other baselines,
+even when the ESBN has observed only a small subset of symbols
+during training.
+
+#### Same-Different Task: Generalization
+{% include 2021-12-01-emergent-symbols/3D_same_diff_trend_subplot.html %}
+In this plot we show the keys written by the ESBN network ($k_w$) at $t=1$, and how it compares to
+the key that the network reads after seeing the second input at $t=2$. The LSTM predicts whether the 
+input at $t=2$ is same as or different from the input at $t=1$ using the input $k_{r_2}$ and the hidden
+state $h_1$. The figure shows how representations of $k_{r_2}$ differ based on whether the example is 
+**Same** or **Different**, showing how ESBN creates a decision boundary for these two cases. Further, 
+the strong clustering for these two cases show that ESBN learns instance invariant representations as claimed in the paper.
+This generalisation is observed for both test and train examples, and the representations align closely.
+
+#### RMTS Task: Generalization
+{% include 2021-12-01-emergent-symbols/3D_RMTS_trend_subplot.html %}
+For the **RMTS** task, we plot the representations for $k_{r}$ for $t= 3$ and $4$ denoting the two possible cases
+for each example. The output of ESBN model would depend on whether the inputs at $t= 3$ and $4$ follow the same relationship as inputs at 
+$t= 1$ and $2$. This can be observed in the segregation between *same relation* (green and orange) and *different relation*
+(purple and sky blue) $k_r$ representations. For the *same relation* case, 
+the test representations align well with the train representations. However, when there is a  *different relation*,
+the test representations seem to interpolate between the clusters formed by the train representations.
+
+
+#### Distribution of Three Task: Generalization
+{% include 2021-12-01-emergent-symbols/3D_dist3_trend_subplot.html %}
+For the **Distribution of Three** task, the first three inputs uniquely determine the symbols for the next three time steps. 
+We plot the retrieved representations ($k_r$) for the first match of the corresponding input at $t = $ 1, 2 and 3. In the plot, 
+we observe that the test representation closely matches with the train representation within the expected range of the training examples.
+However, there are lot of outlier data points in the test representation.
+The generalisation of performance on the test distribution is not perfect,
+and the representations do not align well with the train distribution.
+In the appendix of the paper, the representations for the
+test distribution are shown only in range of the train representation. When observed in this range, the train and test representations are well aligned. However, there are some outlier test instances which outside the train representations range showing that the alignment is not perfect.
+
+<!-- We observe that the alignment between train and test representations is not perfect. -->
```

Comment:
Could you remove this comment? It messes with the excerpt rendering

Reply:
Done!

Reply:
Thanks for adding the content! I think it needs to be under the `excerpt` tag as in the example above (not `abstract`), sorry for not making that clear